odoo/addons/base/models/assetsbundle.py

```diff
@@ -278,7 +278,6 @@ def save_attachment(self, type, content):
         mimetype = 'application/javascript' if type == 'js' else 'text/css'
         values = {
             'name': "/web/content/%s" % type,
-            'datas_fname': fname,
```

Comment:
here ´name´ and ´datas_fname´ doesn't contain the same value ; could be an issue that we loose information.

Reply:
@odony you maybe want to review the small chuncks in this file `assetsbundle.py`. The rest seems safe.

Reply:
Indeed this will break the deduplication mechanism of assets during concurrent generation.

Reply:
To be more explicit about this case, the situation is actually explained in the comment [at the top of the method](https://github.com/odoo/odoo/blob/1dfc45aa8817becca4928db263dd299b9a955210/odoo/addons/base/models/assetsbundle.py#L243-L248). You can reproduce it fairly easily, by simulating a case where we update the source code and restart the server:
- drop a bundle attachment (or several) in the database to simulate a missing one (e.g new one added by the update)
- restart the server to clear the caches
- now load `/web` multiple times in parallel, simulating users reconnecting after the server restart
- result: multiple copies of the new bundle are now cached, with different IDs (each transaction could not see that another was generating the same bundle) 
- finally refresh /web and look at the source of the page - you will see that the bundle(s) are now sourced multiple times, causing random problems with CSS and JS. The bundle system can split files in multiple attachments, and it does not know that those are the same.
![Screenshot from 2019-06-05 03-26-31](https://user-images.githubusercontent.com/2039870/58923855-c3da5300-8741-11e9-8ee3-73d1956f8e89.png)

In order to fix this case I propose to do this simple patch: 7ff28b9439d